src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -57,6 +60,53 @@ def handleRequest(request)
         response["X-Puppet-Version"])
   end
 
+  def getClassInfoForAllEnvironments()
+    # The clear_environment_settings(env) ensures that the calls which happen
+    # later on to enumerate all of the manifests for each environment are
+    # using the latest environment.conf settings for each environment.  Without
+    # this call, cached (and, therefore, possibly stale) environment.conf
+    # settings would be used.  We want to return the latest data for any calls
+    # made to this method.
+    @env_loader.list.each do |env|
+      Puppet.settings.clear_environment_settings(env)
+    end
+
+    environments =
+      Hash[@env_loader.list.collect do |env|
+       [env.name, self.class.getManifests(env)]
+      end]
+
+    classes_per_env =
+        Puppet::InfoService::ClassInformationService.new.classes_per_environment(environments)
+    Hash[classes_per_env.collect {|key, value| [key.to_s, value]}]
```

Comment:
@dankreek - Here's that logic I mentioned to you where I decoded the environment Symbol that appears as the key for the outermost Hash back into a String before returning back across the Java interop layer.  I did this here because I was thinking that the consumers of this in Clojure would likely need to reference elements from the HashMap by environment key but likely would treat anything in the value per environment as a more opaque blob - handed off to cheshire or some other JSON serializer as needed.  This could help keep the Clojure-level handling of the return structure a little more generic - i.e., not having to know that the environment was a JRuby `RubySymbol` before handling it.   The "all environments by key, without serialization" integration test in "class_info_test.clj" which is associated with this validates that a Clojure consumer would only need to do a simple `get` call to access the class info for a specific environment.  Still open to doing this differently.